### PR TITLE
A little bit of documentation for how to use ReactContext

### PR DESCRIPTION
**Pull request by rleibman**
This is actually working on my application, I thought I'd save others the hassle of figuring it out. Let me know if you think I got it right.

### CODE REVIEW

**Comment by japgolly** [comment]
Thanks a lot @rleibman for doing this! More documentation is always appreciated.

Just one change I request, could you change the code snippet to use the newer methods on `Ajax` so that the return type becomes `AsyncCallback` instead of `Future[CallbackTo[...]]`?

**Comment by rleibman** [comment]
Sorry, I haven't yet used your Ajax call, it's missing some of the stuff I need: put (Yeah, I know I can force the method) and credentials. I'll take a look and make the change in the doc.

**Comment by rleibman** [comment]
Just trying it now... I'm actually quite confused as to how to replace the original Ajax with yours.
Previously, this worked:
```scala
for {
   a1 <- Ajax... (returning Future[CallbackTo]]
  a2 <- Ajax...(returning Future[CallbackTo]]
} yield(a1 << a2)
```
But when I start using
```scala
Ajax("GET", s"http://helloworld")
        .setRequestHeader("Content-Type", "application/json")
        .and(_.withCredentials = true)
        .send(someJson)
           .onComplete { xhr ⇒
            val json = xhr.responseText
            CallbackTo(read[Seq[ScheduledMeal]](json))
          }
```
the don't compose, and though I can call .asCallback on the result of that, that doesn't give me back my a1, a2

**Comment by rleibman** [comment]
ok, I made the change. I'm thinking of moving all of my code to use your ajax calls, but it seems daunting!

File: doc/CONTEXT.md [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+        .and(_.withCredentials = true)
+        .send
+        .asAsyncCallback
+        .map { xhr ⇒
+          try {
+            import model.ModelPickler._
+            import upickle.default._
+            val user = Option(read[User](xhr.responseText))
+            //This should cause a cascade rendering all the way down the component stack
+            $.modState(s => s.copy(myGlobalState = s.myGlobalState.copy(user = user)))
```
Comment (japgolly) [inline doc/CONTEXT.md]:
```suggestion
            $.modState(s => s.copy(myGlobalState = s.myGlobalState.copy(user = user)))
              .asAsyncCallback
```

File: doc/CONTEXT.md [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+            val user = Option(read[User](xhr.responseText))
+            //This should cause a cascade rendering all the way down the component stack
+            $.modState(s => s.copy(myGlobalState = s.myGlobalState.copy(user = user)))  
+          } catch {
+            case e: InvalidData ⇒
+              dom.console.error(e.msg + ":" + e.data)
+              throw e
+          }
+        }
+        .completeWith(_.get)
```
Comment (japgolly) [inline doc/CONTEXT.md]:
```suggestion
        .toCallback
```

File: doc/CONTEXT.md [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+      MyGlobalState.ctx.provide(s.myGlobalState) {
+        <.div(AppRouter.router())
+      }
+
+    def refresh(s: State) =
+      Ajax.get(s"https://localhost:8000/whoami")
+        .and(_.withCredentials = true)
+        .send
+        .asAsyncCallback
+        .map { xhr ⇒
```
Comment (japgolly) [inline doc/CONTEXT.md]:
```suggestion
        .flatMap { xhr ⇒
```
^^ This too. Nearly forgot.

**Comment by japgolly** [comment]
Merged. Thanks a lot for your contribution @rleibman !

**Pull request merged**